CSNotificationView/CSNotificationView.m

```diff
@@ -383,6 +383,10 @@ - (void)dismissWithStyle:(CSNotificationViewStyle)style message:(NSString *)mess
 - (CGFloat)topLayoutGuideLengthCalculation
 {
     CGFloat top = MIN([UIApplication sharedApplication].statusBarFrame.size.height, [UIApplication sharedApplication].statusBarFrame.size.width);
+
+    UIView *view = self.parentNavigationController.view ?: self.parentViewController.view;
+    CGPoint origin = [view.superview convertPoint:view.frame.origin toView:[UIApplication sharedApplication].keyWindow.rootViewController.view];
```

Comment:
OK, now I understand what you are trying to achieve.
However: You are mixing visual impression ("being at the top of the screen") with the logical representation in Apple's frameworks. This is a little too hacky for me and it looks like it can easily break.

`topLayoutGuideLengthCalculation` tries to calculate the vertical indentation from 0 to y in the `parentViewController` or `parentNavigationController` because `topLayoutGuide` is not always reliable / useful (e.g. when presenting in a `UINavigationController`). Hence, the name of the method.

In you case (modal popover on iPad), using the parent's layout guide is enough.
```
return [self.parentViewController.topLayoutGuide length];
````

However, this does not work as a general solution (try yourself ;) ).

ATM, I don't have a satisfying answer to this. In principle, we need a way to express the condition of being in a modal popover via API calls (on iOS 7 and iOS 8).

Do you have any idea?